servers/lib/src/files/git/git-files.service.ts

```diff
@@ -1,20 +1,66 @@
-import { Injectable } from '@nestjs/common';
+import { Inject, Injectable, Logger } from '@nestjs/common';
 import { Project } from 'src/types.js';
 import { IFilesService } from '../interfaces/files.service.interface.js';
 import { CONFIG_MODE } from '../../enums/config-mode.enum.js';
+import * as git from 'isomorphic-git';
+import * as fs from 'fs';
+import * as http from 'isomorphic-git/http/node/index.cjs';
+import LocalFilesService from '../local/local-files.service.js';
+import Config from '../../config/config.service.js';
+import { CONFIG_SERVICE } from '../../config/config.interface.js';
+import path from 'path';
 
 @Injectable()
 export default class GitFilesService implements IFilesService {
-// eslint-disable-next-line no-useless-constructor, no-empty-function
-  constructor() {}
+  private readonly dataPath: string;
+  private readonly logger: Logger;
+  @Inject(LocalFilesService) private localFilesService: LocalFilesService;
+
+  constructor(@Inject(CONFIG_SERVICE) private configService: Config) {
+    this.dataPath = this.configService.getLocalPath();
+    this.logger = new Logger(GitFilesService.name);
+  }
+
+  async init(): Promise<void> {
+    await this.cloneRepositories();
```

Comment:
The await makes this a synchronous call

Reply:
@prasadtalasila because it is one threaded or why?